example/exampleMapD3Crossfilter.html

```diff
@@ -0,0 +1,745 @@
+      binExtent: [],
+      rangeBrushExtent: [],
+      timeBin: null,
+      extract: null,
+      autoBin: null,
+      brushExtent: [],
+      keyType: "time",
+      yTicks: "auto",
+      xDomain: "auto",
+      yDomain: "auto",
+      y2Domain: "auto",
+      xLock: false,
+      yLock: false,
+      y2Lock: false,
+      hasRightAxis: false,
+      hasLeftAxis: true,
+      yAxisFormat: "auto",
+      y2AxisFormat: "auto",
+      tooltipFormat: "auto",
+      tooltipTitleFormat: "auto",
+      xAxisFormat: "auto",
+      measureFormats: [],
+      dimensionFormats: []
+    })
+    .setData(dataTransformed)
+  }
+
+  // date object formatter for constructing SQL query from combo chart brushing
+  const dateFormatter = mapd3.d3.timeFormat("%Y-%m-%d %H:%M:%S")
+
+  // set up a few event listeners for the comboChart's brush events
+  comboChart
+    // we can access the combo chart's event listeners using .getEvents()
+    .getEvents()
+    // note: it's a good idea to "throttle" the brushMove event as it gets called very frequently
+    .onBrush("brushMove", throttle(function() {
+      brushExtent = [].concat(arguments[0])
+      adapter.filter(brushExtent, "auto", false)
+      adapter.redrawGroup();
+    }, 100))
+    .onBrush("brushEnd", function() {
+      brushExtent = [].concat(arguments[0])
+      adapter.filter(brushExtent, "auto", false)
+      adapter.redrawGroup();
+    })
+    .onBrush("brushClear", function() {
+      brushExtent = []
+      adapter.filter()
+      adapter.redrawGroup();
+    });
+
+
+    // make the initial query and render the combo chart
+    queryDB(query).then(data => {
+      renderComboChart(data)
+    })
+
+    // render the DC Charts
+    dc.renderAllAsync()
+
+    /*--------------------------RESIZE EVENT------------------------------*/
+
+    /* Here we listen to any resizes of the main window.  On resize we resize the corresponding widgets and call dc.renderAll() to refresh everything */
+
+    window.addEventListener("resize", debounce(reSizeAll, 100));
+
+    function reSizeAll(){
+      var w = Math.max(document.documentElement.clientWidth, window.innerWidth || 0) - 50
+      var h = Math.max(document.documentElement.clientHeight, window.innerHeight || 0) - 200
+
+      dcBarChart
+        .height(h/1.5)
+        .width(w/2)
+
+      dcScatterPlot
+        .height(h/1.5)
+        .width(w/2)
+
+      comboChart
+        .setConfig({
+          width: w
+        })
+
+      dc.redrawAllAsync();
+      comboChart.render()
+    }
+  }
+
+
+  function debounce(func, wait, immediate) {
+    var timeout;
+    return function() {
+      var context = this, args = arguments;
+      var later = function() {
+        timeout = null;
+        if (!immediate) func.apply(context, args);
+      };
+      var callNow = immediate && !timeout;
+      clearTimeout(timeout);
+      timeout = setTimeout(later, wait);
+      if (callNow) func.apply(context, args);
+    };
+  };
+
+  function throttle(func, limit) {
```

Comment:
We could expose the throttle utility function that is in mapd3 src/charts/helpers/common.js.

Reply:
hmmm, we could but it seems that might be going towards the direction of making mapd3 more of a general purpose JS library then a declarative way of creating charts?